docs/getting_started.rst

```diff
@@ -1,135 +1,288 @@
-large, often too large to be held in a single machine but for demonstration
-purposes, we use a small DataFrame with a mixed set of types.
+
+``kartothek`` manages datasets that consist of files that contain tables. It does so by offering
+a metadata definition to handle these datasets efficiently.
+
+Datasets in ``kartothek`` are made up of one or more ``tables``, each with a unique schema.
+When working with ``kartothek`` tables as a Python user, we will use :class:`~pandas.DataFrame`
+as the user-facing type.
+
+We typically expect that the contents of a dataset are
+large, often too large to be held in memory by a single machine but for demonstration
+purposes, we will use a small DataFrame with a mixed set of types.
 
 .. ipython:: python
 
-   import numpy as np
-   import pandas as pd
-
-   df = pd.DataFrame(
-       {
-           "A": 1.,
-           "B": pd.Timestamp("20130102"),
-           "C": pd.Series(1, index=list(range(4)), dtype="float32"),
-           "D": np.array([3] * 4, dtype="int32"),
-           "E": pd.Categorical(["test", "train", "test", "train"]),
-           "F": "foo",
-       }
-   )
-   df
+    import numpy as np
+    import pandas as pd
+
+    df = pd.DataFrame(
+        {
+            "A": 1.0,
+            "B": pd.Timestamp("20130102"),
+            "C": pd.Series(1, index=list(range(4)), dtype="float32"),
+            "D": np.array([3] * 4, dtype="int32"),
+            "E": pd.Categorical(["test", "train", "test", "train"]),
+            "F": "foo",
+        }
+    )
+    df
+
+
+Defining the storage location
+=============================
 
 We want to store this DataFrame now as a dataset. Therefore, we first need
-to connect to a storage location. ``kartothek`` can write to any location that
-fulfills the `simplekv.KeyValueStore interface`_. We use `storefact`_ in this
-example to construct such a store for the local filesystem.
+to connect to a storage location.
+
+We define a store factory as a callable which contains the storage information.
+We will use `storefact`_ in this example to construct such a store factory
+for the local filesystem (``hfs://`` indicates we are using the local filesystem and
+what follows is the filepath).
 
 .. ipython:: python
 
-   from storefact import get_store_from_url
-   from tempfile import TemporaryDirectory
+    from functools import partial
+    from tempfile import TemporaryDirectory
+    from storefact import get_store_from_url
+
+    dataset_dir = TemporaryDirectory()
+
+    store_factory = partial(get_store_from_url, f"hfs://{dataset_dir.name}")
 
-   dataset_dir = TemporaryDirectory()
-   store = get_store_from_url(f"hfs://{dataset_dir.name}")
 
-Now that we have our data and the storage location, we can persist it to a dataset.
-For that we use in this guide :func:`kartothek.io.eager.store_dataframes_as_dataset`
-to store a ``DataFrame`` we already have in memory in the local task.
+For a more technical description of storage specification in kartothek, see
```

Comment:
What kind of technical description? Does this link exist, yet? What would you want to write down there? Instead of deferring to another page, I'd create a text box which describes that we're using the simplekv interface and the user should ensure that the store he is using is an instance of `ExtendedKeyspaceMixin`, achieved by using the `h-` prefix when using storefact

Reply:
Yes, it's linked to the Storage Specification section in the Further Useful Features guide.

Reply:
I agree with recommending to use the `h-` prefix in storefact, but why bother a new user with technical requirements of the store (`ExtendedKeyspaceMixin` et al)?